File: app/src/main/java/com/wenitech/cashdaily/ActivityMain/ActivityClientes/Fragment/VencidosFragment.java
package com.wenitech.cashdaily.ActivityMain.ActivityClientes.Fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.firebase.ui.firestore.FirestoreRecyclerOptions;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.firestore.CollectionReference;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.Query;
import com.wenitech.cashdaily.Adapter.RecyclerViewClienteAdapter;
import com.wenitech.cashdaily.Model.Cliente;
import com.wenitech.cashdaily.R;

public class VencidosFragment extends Fragment {

    private FirebaseAuth mAuth = FirebaseAuth.getInstance();
    private FirebaseFirestore db = FirebaseFirestore.getInstance();
    private CollectionReference ClientesRef = db.collection("usuarios").document(mAuth.getUid()).collection("clientes");

    private RecyclerViewClienteAdapter mRecyclerviewClienteAdapter;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_vencidos,container,false);

        Query query = ClientesRef.orderBy("nombreCliente",Query.Direction.ASCENDING);
        FirestoreRecyclerOptions<Cliente> options = new FirestoreRecyclerOptions.Builder<Cliente>()
                .setQuery(query,Cliente.class).build();

        RecyclerView mRecyclerView = view.findViewById(R.id.recyclerView);
        mRecyclerviewClienteAdapter = new RecyclerViewClienteAdapter(options);
        mRecyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
        mRecyclerView.setAdapter(mRecyclerviewClienteAdapter);
        return view;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public void onStart() {
        super.onStart();
        mRecyclerviewClienteAdapter.startListening();
    }

    @Override
    public void onStop() {
        super.onStop();
        mRecyclerviewClienteAdapter.stopListening();
    }
}

File: app/src/main/java/com/wenitech/cashdaily/features/home/ui/VencidosFragment.kt
package com.wenitech.cashdaily.features.home.ui

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.firebase.ui.firestore.FirestoreRecyclerOptions
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.FirebaseFirestore
import com.google.firebase.firestore.Query
import com.wenitech.cashdaily.R
import com.wenitech.cashdaily.data.model.Cliente
import com.wenitech.cashdaily.features.client.ui.adapter.RecyclerViewClienteAdapter

class VencidosFragment : Fragment(), RecyclerViewClienteAdapter.ReciclerViewClienteInteface {

    private val mAuth = FirebaseAuth.getInstance()
    private val db = FirebaseFirestore.getInstance()

    private val ClientesRef = db.collection("usuarios").document(mAuth.uid!!).collection("clientes")
    private var mRecyclerviewClienteAdapter: RecyclerViewClienteAdapter? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {

        val view = inflater.inflate(R.layout.fragment_vencidos, container, false)
        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val query = ClientesRef.orderBy("fullName", Query.Direction.ASCENDING)
        val options = FirestoreRecyclerOptions.Builder<Cliente>().setQuery(query, Cliente::class.java).build()

        val mRecyclerView: RecyclerView = view.findViewById(R.id.recyclerView)
        mRecyclerviewClienteAdapter = RecyclerViewClienteAdapter(options, this)
        mRecyclerView.layoutManager = LinearLayoutManager(context)
        mRecyclerView.adapter = mRecyclerviewClienteAdapter
    }

    override fun onStart() {
        super.onStart()
        mRecyclerviewClienteAdapter!!.startListening()
    }

    override fun onStop() {
        super.onStop()
        mRecyclerviewClienteAdapter!!.stopListening()
    }

    override fun onClientClicked(cardView: View?, cliente: Cliente?) {
        Toast.makeText(requireContext(), cliente!!.fullName.toString(), Toast.LENGTH_SHORT).show()
    }
}